src/entities.rs

```diff
@@ -27,26 +27,26 @@ impl Entity {
     ///
     /// No particular structure is guaranteed for the returned bits.
     ///
+    /// If bits does not correspond to a possible `NonZeroU64` value returns `None`
+    ///
     /// Useful for storing entity IDs externally, or in conjunction with `Entity::from_bits` and
     /// `World::spawn_at` for easy serialization. Alternatively, consider `id` for more compact
     /// representation.
-    pub fn to_bits(self) -> u64 {
-        NonZeroU64::from(self.generation).get() << 32 | u64::from(self.id)
+    pub fn to_bits(self) -> Option<NonZeroU64> {
+        NonZeroU64::new(NonZeroU64::from(self.generation).get() << 32 | u64::from(self.id))
     }
 
-    /// Reconstruct an `Entity` previously destructured with `to_bits`
+    /// Reconstruct an `Entity` previously destructured with `to_bits` if the `bits` is a value
+    /// that can correspond to possible `NonZeroU64`
+    /// 
     ///
     /// Useful for storing entity IDs externally, or in conjunction with `Entity::to_bits` and
     /// `World::spawn_at` for easy serialization.
     pub fn from_bits(bits: u64) -> Option<Self> {
-        if NonZeroU32::new((bits >> 32) as u32).is_some() {
-            Some(Self {
-                generation: NonZeroU32::new((bits >> 32) as u32).unwrap(),
-                id: bits as u32,
-            })
-        } else {
-            None
-        }
+        NonZeroU32::new((bits >> 32) as u32).map(|_| Self {
```

Comment:
Now that to_bits returns Option<NonZeroU64> shouldn't we assume that the user will deal with the option, and just use new_unchecked here ?

Reply:
Using `new_unchecked` here would be unsound, because the user could pass in 0.